src/main/java/luke/TaskList.java

```diff
@@ -0,0 +1,229 @@
+                return false;
+            }
+            String startDate = StringManipulation.getFirstDetail(taskDate);
+            String endDate = StringManipulation.removeFirstDetail(taskDate);
+
+            if (startDate == null || endDate == null) {
+                return false;
+            }
+
+            newTask = new Event(taskName, newTaskID, startDate.trim(), endDate.trim());
+            break;
+        }
+        tasks.put(newTaskID, newTask);
+        newTaskID += 1;
+        updateSerialNumbers();
+        return true;
+    }
+
+    /**
+     * Given the serial number of a task, this function marks the corresponding task as completed.
+     *
+     * @param toMarkSerialNumber The serial number of the task to be marked.
+     */
+    public void markTask(int toMarkSerialNumber) {
+        int toMarkID = serialNumbers.get(toMarkSerialNumber);
+        tasks.get(toMarkID).markTask();
+    }
+
+    /**
+     * Given the serial number of a task, this function unmarks a task.
+     *
+     * @param toUnmarkSerialNumber The serial number of the task to be unmarked.
+     */
+    public void unmarkTask(int toUnmarkSerialNumber) {
+        int toUnmarkID = serialNumbers.get(toUnmarkSerialNumber);
+        tasks.get(toUnmarkID).unmarkTask();
+    }
+
+    /**
+     * Given the ID of a task, this function deletes a task.
+     *
+     * @param toDeleteSerialNumber The ID of the task to be deleted.
+     */
+    public void deleteTask(int toDeleteSerialNumber) {
+        int toDeleteID = serialNumbers.get(toDeleteSerialNumber);
+        tasks.remove(toDeleteID);
+        updateSerialNumbers();
+    }
+
+    public ArrayList<Task> findTask(String toFind) {
+        ArrayList<Task> found = new ArrayList<>();
+        for (Map.Entry<Integer, Task> entry : tasks.entrySet()) {
+            Task task = entry.getValue();
+            if (task.getTaskName().contains(toFind)) {
+                found.add(task);
+            }
+        }
+        return found;
+    }
+
+    /**
+     * Returns the name of a task, given its serial number.
+     *
+     * @param taskSerialNumber The serial number of the task.
+     * @return The name of the task with taskSerialNumber.
+     */
+    public String getTaskBySerialNumber(int taskSerialNumber) {
+        int taskID = serialNumbers.get(taskSerialNumber);
+        return tasks.get(taskID).getTaskName();
+    }
+
+    /** Updates the serialNumbers to match the current IDs of the tasks */
+    private void updateSerialNumbers() {
+        int serialNumber = 1;
+        for (Map.Entry<Integer, Task> entry : tasks.entrySet()) {
+            Task task = entry.getValue();
+            serialNumbers.put(serialNumber, task.getTaskID());
+            serialNumber += 1;
+        }
+    }
+
+    /**
+     * Returns an ArrayList of Tasks in the order they are added.
+     *
+     * @return ArrayList of Tasks.
+     */
+    public ArrayList<Task> getTaskList() {
+        ArrayList<Task> copy = new ArrayList<Task>();
```

Comment:
Jave coding standard: Use plural form for collection of objects.

As mentioned above, can give a suitable name in plural form.